### ISSUE TITLE
libnsl.so no longer available on RHEL 8

### ISSUE THREAD

**Issue by MichaelTiemannOSC** [conda-forge/rasterio-feedstock#220]
- [X ] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue:
When I attempt to import rasterio in a Jupyter notebook running on Open Shift in a RHEL 8 environment, it fails because libnsl.so is not found (in library _base.cpython-38-x86_64-linux-gnu.so).  libnsl.so has been deprecated for years and was removed from RHEL 8 (and Fedora 27 I believe) in 2018.  There are many reports across many packages that libnsl.so is gone and not coming back.  I'm reporting the fact that it affects rasterio and I think it's because of the way you have build your cpython dependencies.

<br/>
Environment (<code>conda list</code>):
<details>

```
# packages in environment at /opt/app-root/miniconda3:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                      1_llvm    conda-forge
affine                    2.3.0                      py_0    conda-forge
argon2-cffi               20.1.0           py38h497a2fe_2    conda-forge
async_generator           1.10                       py_0    conda-forge
attrs                     21.2.0             pyhd8ed1ab_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
bleach                    4.1.0              pyhd8ed1ab_0    conda-forge
boost-cpp                 1.74.0               h312852a_4    conda-forge
branca                    0.4.2              pyhd8ed1ab_0    conda-forge
brotlipy                  0.7.0           py38h27cfd23_1003  
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.17.2               h7f98852_0    conda-forge
ca-certificates           2021.5.30            ha878542_0    conda-forge
cairo                     1.16.0            h6cf1ce9_1008    conda-forge
certifi                   2021.5.30        py38h578d9bd_0    conda-forge
cffi                      1.14.6           py38h400218f_0  
cfitsio                   3.470                hb418390_7    conda-forge
chardet                   4.0.0           py38h06a4308_1003  
click                     7.1.2              pyh9f0ad1d_0    conda-forge
click-plugins             1.1.1                      py_0    conda-forge
cligj                     0.7.2              pyhd8ed1ab_0    conda-forge
conda                     4.10.3           py38h578d9bd_2    conda-forge
conda-package-handling    1.7.3            py38h27cfd23_1  
cryptography              3.4.7            py38hd23ed53_0  
curl                      7.79.1               hea6ffbf_0    conda-forge
cycler                    0.10.0                     py_2    conda-forge
debugpy                   1.4.1            py38h709712a_0    conda-forge
decorator                 4.4.2                      py_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
entrypoints               0.3             pyhd8ed1ab_1003    conda-forge
expat                     2.4.1                h9c3ff4c_0    conda-forge
fiona                     1.8.20           py38hbb147eb_1    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 hab24e00_0    conda-forge
fontconfig                2.13.1            hba837de_1005    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
freetype                  2.10.4               h0708190_1    conda-forge
freexl                    1.0.6                h7f98852_0    conda-forge
gdal                      3.3.2            py38h81a01a0_2    conda-forge
geopandas                 0.9.0              pyhd8ed1ab_1    conda-forge
geopandas-base            0.9.0              pyhd8ed1ab_1    conda-forge
geos                      3.9.1                h9c3ff4c_2    conda-forge
geotiff                   1.7.0                h4f31c25_0    conda-forge
gettext                   0.19.8.1          h0b5b191_1005    conda-forge
giflib                    5.2.1                h36c2ea0_2    conda-forge
hdf4                      4.2.15               h10796ff_3    conda-forge
hdf5                      1.12.1          nompi_h2750804_100    conda-forge
icu                       68.1                 h58526e2_0    conda-forge
idna                      2.10               pyhd3eb1b0_0  
importlib-metadata        4.8.1            py38h578d9bd_0    conda-forge
ipykernel                 6.4.1            py38he5a9106_0    conda-forge
ipyleaflet                0.14.0             pyhd8ed1ab_1    conda-forge
ipython                   7.27.0           py38he5a9106_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.6.5              pyhd8ed1ab_0    conda-forge
jbig                      2.1               h7f98852_2003    conda-forge
jedi                      0.18.0           py38h578d9bd_2    conda-forge
jinja2                    3.0.1              pyhd8ed1ab_0    conda-forge
joblib                    1.0.1              pyhd8ed1ab_0    conda-forge
jpeg                      9d                   h36c2ea0_0    conda-forge
json-c                    0.15                 h98cffda_0    conda-forge
jsonschema                3.2.0              pyhd8ed1ab_3    conda-forge
jupyter_client            7.0.3              pyhd8ed1ab_0    conda-forge
jupyter_core              4.8.1            py38h578d9bd_0    conda-forge
jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
jupyterlab_widgets        1.0.2              pyhd8ed1ab_0    conda-forge
kealib                    1.4.14               h87e4c3c_3    conda-forge
kiwisolver                1.3.2            py38h1fd1430_0    conda-forge
krb5                      1.19.2               hcc1bbae_0    conda-forge
lcms2                     2.12                 hddcbb42_0    conda-forge
ld_impl_linux-64          2.35.1               h7274673_9  
lerc                      2.2.1                h9c3ff4c_0    conda-forge
libblas                   3.9.0           11_linux64_openblas    conda-forge
libcblas                  3.9.0           11_linux64_openblas    conda-forge
libcurl                   7.79.1               h2574ce0_0    conda-forge
libdap4                   3.20.6               hd7c4107_2    conda-forge
libdeflate                1.7                  h7f98852_5    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libffi                    3.3                  he6710b0_2  
libgcc-ng                 11.2.0               h1d223b6_8    conda-forge
libgdal                   3.3.2                hd2dd7ad_2    conda-forge
libgfortran-ng            11.2.0               h69a702a_8    conda-forge
libgfortran5              11.2.0               h5c6108e_8    conda-forge
libglib                   2.68.4               h3e27bee_0    conda-forge
libiconv                  1.16                 h516909a_0    conda-forge
libkml                    1.3.0             h238a007_1014    conda-forge
liblapack                 3.9.0           11_linux64_openblas    conda-forge
libnetcdf                 4.8.1           nompi_hb3fd0d9_101    conda-forge
libnghttp2                1.43.0               h812cca2_0    conda-forge
libopenblas               0.3.17          pthreads_h8fe5266_1    conda-forge
libpng                    1.6.37               h21135ba_2    conda-forge
libpq                     13.3                 hd57d9b9_0    conda-forge
librttopo                 1.1.0                h1185371_6    conda-forge
libsodium                 1.0.18               h36c2ea0_1    conda-forge
libspatialindex           1.9.3                h9c3ff4c_4    conda-forge
libspatialite             5.0.1                h8694cbe_6    conda-forge
libssh2                   1.10.0               ha56f1ee_1    conda-forge
libstdcxx-ng              11.2.0               he4da1e4_8    conda-forge
libtiff                   4.3.0                hf544144_1    conda-forge
libuuid                   2.32.1            h7f98852_1000    conda-forge
libwebp-base              1.2.1                h7f98852_0    conda-forge
libxcb                    1.13              h7f98852_1003    conda-forge
libxml2                   2.9.12               h72842e0_0    conda-forge
libzip                    1.8.0                h4de3113_1    conda-forge
llvm-openmp               12.0.1               h4bd325d_1    conda-forge
lz4-c                     1.9.3                h9c3ff4c_1    conda-forge
mapclassify               2.4.3              pyhd8ed1ab_0    conda-forge
markupsafe                2.0.1            py38h497a2fe_0    conda-forge
matplotlib-base           3.4.3            py38hf4fb855_0    conda-forge
matplotlib-inline         0.1.3              pyhd8ed1ab_0    conda-forge
mistune                   0.8.4           py38h497a2fe_1004    conda-forge
munch                     2.5.0                      py_0    conda-forge
nb_conda_kernels          2.3.1            py38h578d9bd_0    conda-forge
nbclient                  0.5.4              pyhd8ed1ab_0    conda-forge
nbconvert                 6.2.0            py38h578d9bd_0    conda-forge
nbformat                  5.1.3              pyhd8ed1ab_0    conda-forge
ncurses                   6.2                  he6710b0_1  
nest-asyncio              1.5.1              pyhd8ed1ab_0    conda-forge
networkx                  2.6.3              pyhd8ed1ab_0    conda-forge
notebook                  6.4.4              pyha770c72_0    conda-forge
numpy                     1.21.2           py38he2449b9_0    conda-forge
olefile                   0.46               pyh9f0ad1d_1    conda-forge
openjpeg                  2.4.0                hb52868f_1    conda-forge
openssl                   1.1.1l               h7f98852_0    conda-forge
packaging                 21.0               pyhd8ed1ab_0    conda-forge
pandas                    1.3.3            py38h43a58ef_0    conda-forge
pandoc                    2.14.2               h7f98852_0    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
parso                     0.8.2              pyhd8ed1ab_0    conda-forge
pcre                      8.45                 h9c3ff4c_0    conda-forge
pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    7.2.0            py38h9776b28_2    conda-forge
pip                       21.1.3           py38h06a4308_0  
pixman                    0.40.0               h36c2ea0_0    conda-forge
poppler                   21.09.0              h0e1ea10_2    conda-forge
poppler-data              0.4.11               hd8ed1ab_0    conda-forge
postgresql                13.3                 h2510834_0    conda-forge
proj                      8.0.1                h277dcde_0    conda-forge
prometheus_client         0.11.0             pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.20             pyha770c72_0    conda-forge
pthread-stubs             0.4               h36c2ea0_1001    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pycosat                   0.6.3            py38h7b6447c_1  
pycparser                 2.20                       py_2  
pygments                  2.10.0             pyhd8ed1ab_0    conda-forge
pyopenssl                 20.0.1             pyhd3eb1b0_1  
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
pyproj                    3.2.1            py38h4df08a6_1    conda-forge
pyrsistent                0.17.3           py38h497a2fe_2    conda-forge
pysocks                   1.7.1            py38h06a4308_0  
python                    3.8.10               h12debd9_8  
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python_abi                3.8                      2_cp38    conda-forge
pytz                      2021.1             pyhd8ed1ab_0    conda-forge
pyzmq                     19.0.2           py38ha71036d_2    conda-forge
rasterio                  1.2.8            py38hf34e042_0    conda-forge
readline                  8.1                  h27cfd23_0  
requests                  2.25.1             pyhd3eb1b0_0  
rtree                     0.9.7            py38h02d302b_2    conda-forge
ruamel_yaml               0.15.100         py38h27cfd23_0  
scikit-learn              1.0              py38hacb3eff_1    conda-forge
scipy                     1.7.1            py38h56a6a73_0    conda-forge
send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
setuptools                52.0.0           py38h06a4308_0  
shapely                   1.7.1            py38hb7fe4a8_5    conda-forge
six                       1.16.0             pyhd3eb1b0_0  
snuggs                    1.4.7                      py_0    conda-forge
sqlite                    3.36.0               hc218d9a_0  
terminado                 0.12.1           py38h578d9bd_0    conda-forge
testpath                  0.5.0              pyhd8ed1ab_0    conda-forge
threadpoolctl             2.2.0              pyh8a188c0_0    conda-forge
tiledb                    2.3.4                he87e0bf_0    conda-forge
tk                        8.6.10               hbc83047_0  
tornado                   6.1              py38h497a2fe_1    conda-forge
tqdm                      4.61.2             pyhd3eb1b0_1  
traitlets                 5.1.0              pyhd8ed1ab_0    conda-forge
traittypes                0.2.1              pyh9f0ad1d_2    conda-forge
tzcode                    2021a                h7f98852_2    conda-forge
tzdata                    2021a                he74cb21_1    conda-forge
urllib3                   1.26.6             pyhd3eb1b0_1  
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.36.2             pyhd3eb1b0_0  
widgetsnbextension        3.5.1            py38h578d9bd_4    conda-forge
xerces-c                  3.2.3                h9d8b166_2    conda-forge
xorg-kbproto              1.0.7             h7f98852_1002    conda-forge
xorg-libice               1.0.10               h7f98852_0    conda-forge
xorg-libsm                1.2.3             hd9c2040_1000    conda-forge
xorg-libx11               1.7.2                h7f98852_0    conda-forge
xorg-libxau               1.0.9                h7f98852_0    conda-forge
xorg-libxdmcp             1.1.3                h7f98852_0    conda-forge
xorg-libxext              1.3.4                h7f98852_1    conda-forge
xorg-libxrender           0.9.10            h7f98852_1003    conda-forge
xorg-renderproto          0.11.1            h7f98852_1002    conda-forge
xorg-xextproto            7.3.0             h7f98852_1002    conda-forge
xorg-xproto               7.0.31            h7f98852_1007    conda-forge
xz                        5.2.5                h7b6447c_0  
yaml                      0.2.5                h7b6447c_0  
zeromq                    4.3.4                h9c3ff4c_0    conda-forge
zipp                      3.5.0              pyhd8ed1ab_0    conda-forge
zlib                      1.2.11               h7b6447c_3  
zstd                      1.5.0                ha95c52a_0    conda-forge
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
(base) bash-4.4$ conda info

     active environment : base
    active env location : /opt/app-root/miniconda3
            shell level : 1
       user config file : /opt/app-root/src/.condarc
 populated config files : 
          conda version : 4.10.3
    conda-build version : not installed
         python version : 3.8.10.final.0
       virtual packages : __linux=4.18.0=0
                          __glibc=2.28=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /opt/app-root/miniconda3  (writable)
      conda av data dir : /opt/app-root/miniconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /opt/app-root/miniconda3/pkgs
                          /opt/app-root/src/.conda/pkgs
       envs directories : /opt/app-root/miniconda3/envs
                          /opt/app-root/src/.conda/envs
               platform : linux-64
             user-agent : conda/4.10.3 requests/2.26.0 CPython/3.8.10 Linux/4.18.0-240.23.2.el8_3.x86_64 rhel/8.3 glibc/2.28
                UID:GID : 1000640000:0
             netrc file : None
           offline mode : False
```
</details>

**Comment by xylar** [comment]
I have run into similar troubles (not sure if it was because of rasterio or something else) on a machine I work on.

@ocefpaf, this seems like it might not necessarily be an issue with rasterio itself but rather its dependencies.  Has this come up already in other feedstocks?

**Comment by xylar** [comment]
Likely related:
https://github.com/conda-forge/cmake-feedstock/issues/69
https://github.com/conda-forge/fiona-feedstock/issues/138

**Comment by MichaelTiemannOSC** [comment]
@xylar, thanks fo the comments.  They are related in that anybody who builds a package that accidentally has a reference to libnsl.so is going to fail in this way on RHEL 8 and modern versions of Fedora.  The problem is not with rasterio or I would have posted the bug in the rasterio github.  It's a packaging problem that affects the conda-forge version.  It does not affect the PyPi version, because they packaged their cpython bits better.  Somebody who makes the conda-forge builds must decide: do we care about Red Hat users, and if so, can we please remove this dependency that's been deprecated for many years, and completely gone for several?  I hope so...

**Comment by xylar** [comment]
This sounds like an issue that needs to be discussed by the conda-forge core team (if it hasn't already happened).  I'm not sure this issue will catch their attention.

**Comment by MichaelTiemannOSC** [comment]
I'm definitely out of my element in this brave new CD/CI world.  If this should be handed downstream to conda-forge or upstream to rasterio or sideways to the people who write the recipes for cpython/cython, please convey.  I just assumed that it was the feedstock feeding the forge that is the most likely place to observe the problem in its primordial state.

**Comment by xylar** [comment]
I've posted a message on conda-forge's gitter, so we'll see when (if) someone responds.  https://gitter.im/conda-forge/conda-forge.github.io

**Comment by izahn** [comment]
> When I attempt to import rasterio in a Jupyter notebook running on Open Shift in a RHEL 8 environment, it fails because libnsl.so is not found (in library _base.cpython-38-x86_64-linux-gnu.so). libnsl.so has been deprecated for years and was removed from RHEL 8 (and Fedora 27 I believe) in 2018. There are many reports across many packages that libnsl.so is gone and not coming back. 

I'm not sure the situation is that dire. According to https://forums.fedoraforum.org/showthread.php?325808-How-do-I-get-libnsl-so-1 and https://access.redhat.com/solutions/4975641 libnsl continues to be available as as separate package.

**Comment by xylar** [comment]
@izahn, that's only really a useful solution on machines where one has root access, which seems strongly in contradiction to the philosophy of conda(-forge) to me.

**Comment by izahn** [comment]
@xylar I'm not saying there is no problem, only that the solution can be as simple as packaging `libnsl` in conda-forge and updating affected packages to depend on it. The original report made it sound like `libnsl` was going away entirely, which AFIAK is not the case.

**Comment by MichaelTiemannOSC** [comment]
Let me rephrase: this version of rasterio has no problem: https://pypi.org/project/rasterio/

It has no problem because they recognized that nothing in rasterio has anything whatsoever to do with libnsl.so.1, doesn't need it, and better off with out it.  Right now my work-around is to slam the PyPi version over top of my conda-forge, which is a fragile and bad way to do things.  And loading a stand-alone libnsl.so.1 that should not have been packaged in the first place is equally sloppy.  Can you please take a closer look at why the conda-forge recipes are still referencing this library that they may not actually even use any longer?  Thank you!

**Comment by izahn** [comment]
@MichaelTiemannOSC your original post stated that "libnsl.so has been deprecated for years and was removed from RHEL". That is untrue. In fact `libnsl` is alive and well, and lots of stuff in Fedora (and other distributions) continues to depend on it, including python:
```
$ docker run -ti fedora                                                                       (base) 
Unable to find image 'fedora:latest' locally
latest: Pulling from library/fedora
b9705287bb9f: Pull complete 
Digest: sha256:d18bc88f640bc3e88bbfacaff698c3e1e83cae649019657a3880881f2549a1d0
Status: Downloaded newer image for fedora:latest
[root@719106a48daf /]# dnf repoquery --whatdepends libnsl
Last metadata expiration check: 0:02:20 ago on Sun Sep 26 20:21:06 2021.
sane-backends-0:1.0.32-4.fc34.x86_64
sane-backends-0:1.0.32-5.fc34.x86_64
[root@719106a48daf /]# dnf repoquery --whatdepends libnsl2
Last metadata expiration check: 0:02:28 ago on Sun Sep 26 20:21:06 2021.
am-utils-5:6.2.0-42.fc34.i686
am-utils-5:6.2.0-42.fc34.x86_64
autofs-1:5.1.7-10.fc34.x86_64
autofs-1:5.1.7-17.fc34.x86_64
exim-0:4.94-7.fc34.x86_64
exim-0:4.94.2-2.fc34.x86_64
exim-mon-0:4.94-7.fc34.x86_64
exim-mon-0:4.94.2-2.fc34.x86_64
libnsl2-devel-0:1.3.0-2.fc34.i686
libnsl2-devel-0:1.3.0-2.fc34.x86_64
nss_nis-0:3.1-8.fc34.i686
nss_nis-0:3.1-8.fc34.x86_64
pam-0:1.5.1-5.fc34.i686
pam-0:1.5.1-5.fc34.x86_64
pam-0:1.5.1-6.fc34.i686
pam-0:1.5.1-6.fc34.x86_64
postfix-2:3.5.10-2.fc34.x86_64
postfix-2:3.5.9-7.fc34.x86_64
python2.7-0:2.7.18-11.fc34.i686
python2.7-0:2.7.18-11.fc34.x86_64
python2.7-0:2.7.18-9.fc34.i686
python2.7-0:2.7.18-9.fc34.x86_64
python3-debug-0:3.9.2-1.fc34.i686
python3-debug-0:3.9.2-1.fc34.x86_64
python3-debug-0:3.9.7-1.fc34.i686
python3-debug-0:3.9.7-1.fc34.x86_64
python3-libs-0:3.9.2-1.fc34.i686
python3-libs-0:3.9.2-1.fc34.x86_64
python3-libs-0:3.9.7-1.fc34.i686
python3-libs-0:3.9.7-1.fc34.x86_64
python3.10-0:3.10.0~a6-2.fc34.i686
python3.10-0:3.10.0~a6-2.fc34.x86_64
python3.10-0:3.10.0~rc2-1.fc34.i686
python3.10-0:3.10.0~rc2-1.fc34.x86_64
python3.6-0:3.6.13-2.fc34.i686
python3.6-0:3.6.13-2.fc34.x86_64
python3.6-0:3.6.15-1.fc34.i686
python3.6-0:3.6.15-1.fc34.x86_64
python3.7-0:3.7.10-1.fc34.i686
python3.7-0:3.7.10-1.fc34.x86_64
python3.7-0:3.7.12-1.fc34.i686
python3.7-0:3.7.12-1.fc34.x86_64
python3.8-0:3.8.12-1.fc34.i686
python3.8-0:3.8.12-1.fc34.x86_64
python3.8-0:3.8.8-1.fc34.i686
python3.8-0:3.8.8-1.fc34.x86_64
rwall-0:0.17-58.fc34.x86_64
rwall-server-0:0.17-58.fc34.x86_64
samba-client-libs-2:4.14.2-1.fc34.i686
samba-client-libs-2:4.14.2-1.fc34.x86_64
samba-client-libs-2:4.14.7-0.fc34.i686
samba-client-libs-2:4.14.7-0.fc34.x86_64
sendmail-0:8.16.1-6.fc34.x86_64
sendmail-0:8.16.1-7.fc34.x86_64
slapi-nis-0:0.56.6-2.fc34.x86_64
slapi-nis-0:0.56.7-1.fc34.x86_64
tcp_wrappers-0:7.6-97.fc34.x86_64
tcp_wrappers-libs-0:7.6-97.fc34.i686
tcp_wrappers-libs-0:7.6-97.fc34.x86_64
yp-tools-0:4.2.3-9.fc34.x86_64
ypbind-3:2.7.2-3.fc34.x86_64
ypserv-0:4.1-5.fc34.x86_64
```
The same is true in other Linux distributions, e.g., `python` in Archlinux also depends on `libnls`, as do many other packages, as you can see at https://archlinux.org/packages/core/x86_64/libnsl/.

In your more recent message you not correctly that things depending on `libnsl.so.1` specifically probably doesn't make sense, and I agree with you. Identifying those packages and patching them to look for `libnsl.so` and packaging `libnsl` in conda-forge still makes sense to me.

**Comment by MichaelTiemannOSC** [comment]
Fair enough, I was imprecise about the nature of libnsl.so.1, which WAS removed from glibc, though it lives on in legacy support in the sane packages.  I was taking my information from this section of RHEL documentation: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/considerations_in_adopting_rhel_8/index#compatibility-breaking-changes-in-compilers-and-development-tools_compilers-and-development-tools

And indeed your scanning of Fedora shows that libnsl version 1 is very thinly used (only the sane packages).  Now, if there were a good reason to use libnsl2, then we could have that debate.  But I see you agree with my main point, so I won't press my luck ;-)

**Comment by izahn** [comment]
I submitted a `libnsl` recipe at https://github.com/conda-forge/staged-recipes/pull/16278. I also submitted https://github.com/conda-forge/libuv-feedstock/pull/52 and https://github.com/conda-forge/python-feedstock/pull/507 to build those packages against the new `libnsl` in conda-forge (obviously https://github.com/conda-forge/staged-recipes/pull/16278 needs to be merged first).

We'll see what the community thinks of this and go from there.

**Comment by MichaelTiemannOSC** [comment]
Thank you.  The comment I wrote there really belongs here, so I'm going to move it to here:

This solution is a great example of "be careful what you wish for"! I was really hoping that libnsl would disappear as a dependency for rasterio's cpython routines. Instead, it is now a thing unto itself, to live forever as junk DNA in conda-forge. So be it!

**Comment by izahn** [comment]
No decisions have been made. I've proposed a path forward that basically copies what Fedora and Arch Linux do. If you have a better solution please propose it.

**Comment by mbargull** [comment]
Building `python` with libnsl2 makes sense if that gets the `nis` module working on newer systems.
However, AFAICT, you are rather most likely running into `libxerces-c.so` being linked to `libnsl.so.21`.
See the indirectly vs directly linked libraries:
```
# ldd "${CONDA_PREFIX}"/lib/python*/site-packages/rasterio/_base.*.so | grep libnsl
	libnsl.so.1 => /lib/x86_64-linux-gnu/libnsl.so.1 (0x00007f91c36d1000)
# readelf -d "${CONDA_PREFIX}"/lib/python*/site-packages/rasterio/_base.*.so | grep NEEDED
 0x0000000000000001 (NEEDED)             Shared library: [libgdal.so.29]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
```
(Xerces-C++ is pulled in via GDAL.)
So, you'd want to open a PR against https://github.com/conda-forge/xerces-c-feedstock to get it to link to libnsl2.

**Comment by izahn** [comment]
I believe this is fixed by the most recent python and xerces-c rebuilds. Please test and report back here.

**Comment by ocefpaf** [comment]
I'm closing this but we can re-open if this is not fixed yet. Hopefully we don't need to rebuild gdal :smile:

**Comment by isuruf** [comment]
@izahn, if you have some time, it'd be good to remove `libnsl.so` from our sysroot package so that we don't link to it automatically.
We might have to fix `cmake` and `fiona` first.

**Comment by izahn** [comment]
OK, I'll see what I can do.
